packages/kuma-gui/src/app/common/ResourceStatus.vue

```diff
@@ -33,23 +31,23 @@ import DefinitionCard from './DefinitionCard.vue'
 
 const props = withDefaults(defineProps<{
   total: number
-  online?: number | null
+  online?: number
 }>(), {
-  online: null,
+  online: undefined,
 })
 
 const statusAppearance = computed(() => {
-  if (props.online !== null) {
-    const ratio = props.online / props.total
+  if (typeof props.online === 'undefined') return undefined
 
-    if (ratio <= 0.5) {
-      return 'danger'
-    } else if (ratio < 1) {
-      return 'warning'
-    }
+  const ratio = props.online / props.total
+
+  if (ratio <= 0.5) {
+    return 'danger'
+  } else if (ratio < 1) {
+    return 'warning'
   }
 
-  return null
+  return undefined
```

Comment:
Feels like we could maybe make this a tiny bit better whilst we are here. Feels weird that we need an `else` in here 🤔 

I guess you want to keep your preferred style of exiting early? (I feel its a bit weird that we return `undefined` in two places now!).

Umm, if we make `ratio` a `max` of `1` would we find a compromise here? i.e. keep the exit early style, and remove the double `undefined`? I'm not totally sure myself btw. lmk

Reply:
Explained below, resolving